tools/c7n_azure/c7n_azure/resources/entraid_group.py

```diff
@@ -0,0 +1,518 @@
+
+    def analyze_group_member_types(self, group_id):
+        """Analyze group member types (internal vs external/guest users).
+
+        Required permission: GroupMember.Read.All, User.Read.All
+        """
+        try:
+            # Get group members with userType field explicitly requested
+
+            endpoint = (
+                f'groups/{group_id}/members?$select=id,displayName,'
+                'userPrincipalName,userType'
+            )
+            response = self.make_graph_request(endpoint)
+            members = response.get('value', [])
+
+            has_external_members = False
+            has_guest_members = False
+
+            for member in members:
+                # Only analyze users (not other groups or service principals)
+                if member.get('@odata.type') == '#microsoft.graph.user':
+                    user_type = member.get('userType', 'Member')
+                    user_principal_name = member.get('userPrincipalName', '')
+
+                    # Check if user is a guest
+                    if user_type.lower() == 'guest':
+                        has_guest_members = True
+
+                    # Check if user is external (from different domain)
+                    # External users typically have #EXT# in their UPN or are guests
+                    if '#EXT#' in user_principal_name or user_type.lower() == 'guest':
+                        has_external_members = True
+
+            return {
+                'has_external_members': has_external_members,
+                'has_guest_members': has_guest_members,
+                'total_members': len([
+                    m for m in members
+                    if m.get('@odata.type') == '#microsoft.graph.user'
+                ])
+            }
+        except requests.exceptions.RequestException as e:
+            if "403" in str(e) or "Insufficient privileges" in str(e):
+                log.warning(
+                    f"Insufficient privileges to analyze member types for group {group_id}. "
+                    "Required permissions: GroupMember.Read.All, User.Read.All"
+                )
+
+                return None  # Unknown member types
+            else:
+                log.error(f"Error analyzing member types for group {group_id}: {e}")
+                return None
+
+
+@EntraIDGroup.filter_registry.register('member-count')
+class MemberCountFilter(Filter):
+    """Filter groups based on member count.
+
+    Required permission: GroupMember.Read.All
+
+    :example:
+
+    Find groups with too many members:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: large-groups
+            resource: azure.entraid-group
+            filters:
+              - type: member-count
+                count: 100
+                op: greater-than
+    """
+
+    schema = type_schema(
+        'member-count',
+        count={'type': 'number'},
+        op={'type': 'string', 'enum': ['greater-than', 'less-than', 'equal']}
+    )
```

Comment:
...and use the value filter's schema here.

We can define an annotation key as a way to expose the group's member count in the `resources.json` output. And since the value filter requires a `key`, we can hardcode that to target the annotation key (with awkward nested quotes because of the colon).

```suggestion
    schema = type_schema(
        'member-count',
        rinherit=ValueFilter.schema
    )
  
    annotation_key = 'c7n:MemberCount'

    def __init__(self, data, manager=None):
        data["key"] = f'"{self.annotation_key}"'
        super().__init__(data, manager)
```